### PR TITLE
feat: expose context to debug logger

### PR DESCRIPTION
**Pull request by torkelrogstad**
Implementation of my feature request from #796

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
Thanks for the PR @torkelrogstad! Will take a look shortly 😄

File: options.go [hunk 1]
```diff
@@ -234,9 +234,30 @@ func WithIAMAuthN() Option {
...
 // operations. By default the debug logger is disabled.
+//
+// Deprecated: use WithContextDebugLogger instead
 func WithDebugLogger(l debug.Logger) Option {
+	return func(d *dialerConfig) {
+		d.logger = &debugLoggerWithoutContext{l}
+	}
+}
+
+// WithContextDebugLogger configures a debug lgoger for reporting on internal
```
Comment (jackwotherspoon) [inline options.go:258]: nit: typo
```suggestion
// WithContextDebugLogger configures a debug logger for reporting on internal
```

**Review by jackwotherspoon — APPROVED** [review]
Overall, LGTM. I'll let @enocom also review as he implemented the original debug logging feature.

File: debug/debug.go [hunk 2]
```diff
@@ -14,8 +14,18 @@
 
 package debug
 
+import "context"
+
 // Logger is the interface used for debug logging. By default, it is unused.
+//
+// Deprecated: use ContextLogger instead.
```
Comment (enocom) [inline debug/debug.go:21]: Let's say "Prefer ContextLogger" or similar. We'll keep letting people use Logger if they want.

**Review by enocom — APPROVED** [review]
One minor nit on wording. Otherwise, this looks great. Thanks for this change.

File: debug/debug.go (hunk 2 above)
Comment (enocom) [inline debug/debug.go:21]:
```suggestion
// Prefer ContextLogger instead.
```

**Comment by torkelrogstad** [comment]
Thanks for the quick merge!

**Pull request merged**